drivers/memc/memc_mcux_flexspi.c

```diff
@@ -72,6 +72,32 @@ int memc_flexspi_update_lut(const struct device *dev, uint32_t index,
 	return 0;
 }
 
+int memc_flexspi_update_clock(const struct device *dev,
+		enum memc_flexspi_clock_t clock)
+{
+	struct memc_flexspi_data *data = dev->data;
+
+	/* Wait for bus to be idle before changing flash configuration. */
+	while (!FLEXSPI_GetBusIdleStatus(data->base)) {
+	}
+
+	FLEXSPI_Enable(data->base, false);
+	CLOCK_DisableClock(kCLOCK_FlexSpi);
```

Comment:
You might need to include the clock symbols in this file

```c
#include <fsl_clock.h>
```

Reply:
I need to rework the code. This will not work for all platforms..
I am trying to figure out a way but I am open for any input!

Reply:
One way to solve this may be adding an SOC specific function to support changing the FlexSPI clock dividers. This function could then be implemented in the soc_rt10xx.c file, and be supported for other SOCs using this hyperflash in the future if required.

Reply:
Thanks! I reworked with your suggestions.

Devices that uses hyperflash but do not have an `imxrt_flexspi_clock_set_div` implementation will not build as far as I can tell.